includes/core.php

```diff
@@ -160,6 +162,72 @@ function script_loader_tag( $tag, $handle ) {
 	return $tag;
 }
 
+/**
+ * Asynchronous stylesheet definitions
+ *
+ * Determines which stylesheets should behave
+ * asynchronously on the page by storing their
+ * unique handles in an array.
+ *
+ * @return array
+ */
+function get_known_handles() {
+	return array( 'styles', 'styleguide', 'wp-block-library' );
+}
+
+/**
+ * Add async/defer attributes to enqueued scripts that have the specified script_execution flag.
+ *
+ * @link https://developer.wordpress.org/reference/hooks/style_loader_tag/
+ * @param string $html   The style html output.
+ * @param string $handle The style handle.
+ * @return string
+ */
+function style_loader_tag( $html, $handle ) {
+
+	// Get previously defined stylesheets.
+	$known_handles = get_known_handles();
+
+	// Loop over stylesheets and replace media attribute
+	foreach ( $known_handles as $known_style ) {
+		if ( $known_style === $handle ) {
+			$html = str_replace( "media='all'", "media='print' onload=\"this.media='all'\"", $html );
```

Comment:
@dainemawer Would it make sense to set `noscript` in here. Something like this:

```
// Replace the media all with media print, and add an onload handler to switch it back.
$print_html = str_replace( "media='all'", "media='print' onload='this.media=\"all\"'", $html );

// Wrap the original link in a noscript tag for users who don't have js enabled.
$html = $print_html . '<noscript>' . $html . '</noscript>';
```

Reply:
@dainemawer Thoughts on this comment?

Reply:
I liked this idea - saves us having another function - so consider it added!